core/autoscaling.go

```diff
@@ -334,6 +335,47 @@ func (a *autoScalingGroup) replaceOnDemandInstanceWithSpot(
 	return a.detachAndTerminateOnDemandInstance(odInst.InstanceId)
 }
 
+func (a *autoScalingGroup) getInstanceTypeByTagInASG() (string, error) {
+	svc := a.region.services.autoScaling
+	input := &autoscaling.DescribeTagsInput{
+		Filters: []*autoscaling.Filter{
+			{
+				Name: aws.String("instance-type"),
+			},
+		},
+	}
+	output, err := svc.DescribeTags(input)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case autoscaling.ErrCodeInvalidNextToken:
+				logger.Println(autoscaling.ErrCodeInvalidNextToken, aerr.Error())
+			case autoscaling.ErrCodeResourceContentionFault:
+				logger.Println(autoscaling.ErrCodeResourceContentionFault, aerr.Error())
+			default:
+				logger.Println(aerr.Error())
+			}
+		} else {
+			// Print the error, cast err to awserr.Error to get the Code and
+			// Message from an error.
+			logger.Println(err.Error())
+		}
+		return "", err
+	}
+
+	instanceTypes, err := a.region.requestSpotInstanceTypes()
+	if err != nil {
+		logger.Println(err.Error())
+		return "", err
+	}
+	for _, it := range instanceTypes {
+		if *output.Tags[0].Value == it {
+			return *output.Tags[0].Value, nil
+		}
```

Comment:
This loop is made because some on-demand instance types do not exist as spot instance, is it correct?
Otherwise, I'm not sure to understand it.